### PR TITLE
Fix nil / fixnum comparison

### PR DESCRIPTION
**Pull request by Baltoli**
This is a bit of a weird one - sometimes Companies House doesn't send us the total number of officers that are attached to a company. When this is the case, we see exceptions like `ArgumentError: comparison of Fixnum with nil failed`.

The fix for this is to assume that we have got all the officers with one request. We can't just iterate through until failure because the API responds with `404` when you specify an offset that's too large.

An example company number that causes this issue is `10271424` - note that `total_results` is `null` for this company. I haven't seen any other companies cause this issue.

### CODE REVIEW

**Comment by Baltoli** [comment]
I have submitted a request on the Companies House developer forum asking for clarification on this issue.

File: lib/companies_house/client.rb [hunk 1]
```diff
@@ -38,8 +38,9 @@ def officers(id)
 
       loop do
         page = request(id, '/officers', start_index: offset)
-        total = page['total_results']
         new_items = page['items']
+        total = page['total_results'] || new_items.count
```
Comment (twhaples) [inline lib/companies_house/client.rb]: it's worth inquiring into whether or not a response with nil `total_results` can be trusted to have all the results on the first page

**Comment by Baltoli** [comment]
No reply yet on the forum, but from my manual testing this only seems to occur when there is only one officer attached to the company. This workaround will fix it if this is indeed the case.

**Comment by Baltoli** [comment]
@matt-thomson @twhaples thoughts on this given that this appears to be a pagination bug in the companies house API? No formal confirmation yet as to the source, but this will cause future verifications to potentially become blocked if not fixed / worked around.

**Comment by Baltoli** [comment]
Verbal 👍 from @matt-thomson - will revisit when we get a reply on the forum.

**Comment by twhaples** [comment]
and a late 👍 as well (on the understanding of revisiting)

**Pull request merged**